app/decorators/current_user.rb

```diff
@@ -0,0 +1,46 @@
+class CurrentUser < Draper::Decorator
+  decorates :user
+  delegate_all
+
+  def session_url
+    if object
+      h.link_to("Logout",
+                h.logout_path,
+                method: :delete)
+    else
+      h.link_to("Login with GitHub",
+                "/auth/github",
+                method: :get)
+    end
+  end
+
+  def create_room_url
+    if object && !h.current_page?(h.new_room_path)
+      h.link_to("Create New Room",
+                h.new_room_path)
+    else
+      ""
+    end
+  end
+
+  def form_link(form)
+    if object
+      form.submit("Create room", class: "btn")
+    else
+      h.link_to("Login to create room",
+                "/auth/github",
+                class: "btn")
+    end
+  end
+
+  def close_topic(room)
+    if object == room.user
+      h.button_to("Close this Topic",
+                 h.room_path(room),
+                 method: :put,
+                 class: "btn col s12")
+    else
+      ""
+    end
+  end
+end
```

Comment:
I agree it looks cleaner, but at what value? It doesn't matter for the scope of this project, but if this were going to be something that had to be maintained, it would be a much longer on boarding process to get someone familiar with since it's highly configuration over convention. Plus it was engineered before there was even a problem. I'm fine with it here since I think it's cool you used it and its a great tool to abstract pieces of code, but just something to consider in the future.